shims/net/index.ts

```diff
@@ -61,18 +61,27 @@ export interface SocketDefaults {
   disableSNI: boolean;
 }
 type GlobalOnlyDefaults = 'poolQueryViaFetch' | 'fetchEndpoint' | 'fetchConnectionCache' | 'fetchFunction';
+const neonHostPattern = /^(?:ep-)?([a-z]+)-([a-z]+)-(\w+)/;
```

Comment:
"ep-" part should be mandatory, no?

Reply:
No, if it's old project there is no endpoint.

Reply:
Oh, the ones with project id as first part...
I am afraid that without ep part it's pretty general... wdyt @jawj ?

Reply:
@duskpoet Yeah, false positives (which would screw things up for people using our driver with a non-Neon database) are certainly possible here.

But they should be very rare (because I don't think there are many of these users), and they're easily fixable by overriding the `fetchEndpoint` and/or `wsProxy` configuration options.

So I think this is probably the right balance? I guess we could also check for at least two more parts afterwards, i.e. `/^(?:ep-)?([a-z]+)-([a-z]+)-(\w+)[.][^.]+[.][^.]+/`, but I think this barely helps.

Reply:
Applied the suggestion.